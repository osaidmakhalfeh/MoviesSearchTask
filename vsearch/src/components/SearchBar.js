import React from 'react';

class SearchBar extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = { term: '' };

        // bind handle function to the class
        this.handleMovieName = this.handleMovieName.bind(this);
    }  
  

  onFormSubmit = event => {
    event.preventDefault();
  };

  
  handleMovieName(e){
    this.setState({term: e.target.value}, function () {
        this.props.onSubmit(this.state.term);
    });
  }

  render() {
    return (
      <div className="ui segment">
        <form className="ui form">
          <div className="field">
            <h3>Movies Search</h3>
            <input
              type="text"
              placeholder="Type the name of your favorite movie..."
              value={this.state.term}
              onChange={this.handleMovieName}
            />
          </div>
        </form>
      </div>
    );
  }
}

export default SearchBar;