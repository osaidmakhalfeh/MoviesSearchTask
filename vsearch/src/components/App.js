import React from 'react';
import omdbapi from '../apis/omdbapi';
import SearchBar from './SearchBar';
import MoviesList from './MoviesList';

class App extends React.Component {
  
    state = { movies: [], loading: false };

  onSearchSubmit = async term => {

     this.setState({loading: true});

     await omdbapi.get('', {
      
      // from api docs, s = Movie title to search for, return list.
      params: { s: term } })

      .then(resp => this.setState({movies: resp.data.Search, loading: false}))

      .catch(( error ) => {   // Status code is not 200

        console.log(error);
        this.setState({loading: false});
    }); 
  };

  _renderMoviesList(){
    if(this.state.loading){
        return(
            <div  className="ui active inline loader">
             
            </div>
        );
    }

    if(this.state.movies !== undefined){
        return(
            <MoviesList movies={this.state.movies} />
        );
    }
  }

  render() {
    return (
      <div  className="ui container" style={{ marginTop: '10px' }}>
        <SearchBar id="loadingme" onSubmit={this.onSearchSubmit}/>
        {this._renderMoviesList()}
      </div>
    );
  }
}

export default App;