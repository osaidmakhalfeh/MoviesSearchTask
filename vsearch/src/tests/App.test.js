import './setupTests';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import App from '../components/App';
import SearchBar from '../components/SearchBar';
import Adapter from "enzyme-adapter-react-16";


// This test to check Jest work
describe('Check Jest Work environment', () => {
    it('To Check if Jest work on our project', () => {
        expect(1 + 2).toEqual(3);
     });
  });


// Unit test App component
describe('App component Test', () => {
it('unit test For app componant', () => {
    shallow(<App />);
    });
 });


// Unit test for Search Bar component to make sure itâ€™s rendering
describe('Search Bar Test', () => {
    test("Search Bar", () => {
        const wrapper = shallow(<SearchBar />);
        expect(wrapper.exists()).toBe(true);
    });
});


// Using Snapshots test
describe('Search Bar Test by Enzyme library', () => {
    Enzyme.configure({ adapter: new Adapter() });
    test("SearchBar component", () => {
        const wrapper = shallow(<SearchBar />);
        expect(wrapper).toMatchSnapshot();
    });
});