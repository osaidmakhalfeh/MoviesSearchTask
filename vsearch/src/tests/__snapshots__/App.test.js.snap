// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component 1`] = `ShallowWrapper {}`;

exports[`Search Bar Test by Enzyme library SearchBar component 1`] = `ShallowWrapper {}`;

exports[`SearchBar component 1`] = `ShallowWrapper {}`;
